src/examples/filter-chip/TableWithGlobalFiltersExample.tsx

```diff
@@ -0,0 +1,679 @@
+  }, [getFilteredData, typedChips]);
+
+  const highlightText = (text: string): string => {
+    const ranges = typedChips
+      .flatMap(chip => {
+        const regex = new RegExp(chip, "gi");
+        const matches = [];
+        let match;
+        while ((match = regex.exec(text)) !== null) {
+          matches.push({ start: match.index, end: match.index + match[0].length });
+        }
+        return matches;
+      })
+      .sort((a, b) => a.start - b.start);
+
+    const mergedRanges = ranges.reduce<{ start: number; end: number }[]>((acc, range) => {
+      const last = acc[acc.length - 1];
+      if (!last || last.end < range.start) {
+        acc.push(range);
+      } else {
+        last.end = Math.max(last.end, range.end);
+      }
+      return acc;
+    }, []);
+
+    return (
+      mergedRanges.reduce((acc, range, index) => {
+        const prevEnd = index === 0 ? 0 : mergedRanges[index - 1].end;
+        return (
+          acc +
+          text.slice(prevEnd, range.start) +
+          "<mark>" +
+          text.slice(range.start, range.end) +
+          "</mark>"
+        );
+      }, "") + text.slice(mergedRanges.length ? mergedRanges[mergedRanges.length - 1].end : 0)
+    );
+  };
+
+  return (
+    <>
+      <h3 id="component-table-with-global-filters-example">Table with Global Filters</h3>
+      <GoAContainer mt="m" mb="none">
+        <GoAFormItem label="Filter Results" id="filterChipInput" error={inputError}>
+          <GoABlock gap="xs" direction="row">
+            <GoAInput
+              name="filterChipInput"
+              aria-labelledby="filterChipInput"
+              placeholder="Filter"
+              value={inputValue}
+              maxLength={100}
+              leadingIcon="filter"
+              onChange={handleInputChange}
+              onKeyPress={handleInputKeyDown}
+            />
+            <GoAButton
+              type="primary"
+              onClick={addInputChip}
+              disabled={inputValue === ""}
+              leadingIcon="add">
+              Add
+            </GoAButton>
+          </GoABlock>
+        </GoAFormItem>
+
+        {typedChips.length > 0 && (
+          <div>
+            {typedChips.map((typedChip, index) => (
+              <GoAFilterChip
+                key={index}
+                content={typedChip}
+                mt="s"
+                mr="s"
+                onClick={() => removeTypedChip(typedChip)}
+                error={errorChips.includes(typedChip)}
+              />
+            ))}
+          </div>
+        )}
+
+        <GoATable width="100%">
+          <thead>
+            <tr>
+              {HEADERS.map((header, index) => (
+                <th key={index}>{header}</th>
+              ))}
+            </tr>
+          </thead>
+          <tbody>
+            {dataFiltered.map((item, index) => (
+              <tr key={index}>
+                <td>
+                  <GoABadge type="information" content={item.status} />
```

Comment:
Could not highlight matching filter text inside of Badge content, so created an issue here to deal with that later: https://github.com/GovAlta/ui-components/issues/2374